File: TEST/GO/test_type.go
type SIMPLE struct {
    Uuid gocql.UUID    `db:"Uuid"`;
    Bool bool    `db:"Bool"`;
    Int8 int8    `db:"Int8"`;
    Uint8 uint8    `db:"Uint8"`;
    Int16 int16    `db:"Int16"`;
    Uint16 uint16    `db:"Uint16"`;
    Int32 int32    `db:"Int32"`;
    Uint32 uint32    `db:"Uint32"`;
    Int64 int64    `db:"Int64"`;
    Uint64 uint64    `db:"Uint64"`;
    Float32 float32    `db:"Float32"`;
    Float64 float64    `db:"Float64"`;
    String string    `db:"String"`;
    Name string    `db:"Name"`;
    Date time.Time    `db:"Date"`;
    DateTime time.Time    `db:"DateTime"`;
    Blob [] byte    `db:"Blob"`;
}

// ~~

type COMPOUND struct {
    Id int32    `db:"Id"`;
    Location string    `db:"Location"`;
    Name STRING_STRING_TUPLE    `db:"Name"`;
    NameSet STRING_STRING_TUPLE_SET    `db:"NameSet"`;
    PhoneList STRING_LIST    `db:"PhoneList"`;
    EmailSet STRING_SET    `db:"EmailSet"`;
    CompanyMap STRING_STRING_MAP    `db:"CompanyMap"`;
    SimpleDate time.Time    `db:"SimpleDate"`;
    SimpleDateMap STRING_STRING_TUPLE_TIME.TIME_MAP    `db:"SimpleDateMap"`;
    SimpleDateSet TIME.TIME_SET    `db:"SimpleDateSet"`;
    SimpleDateList TIME.TIME_LIST    `db:"SimpleDateList"`;
    NameSetMap TIME.TIME_STRING_STRING_TUPLE_SET_MAP    `db:"NameSetMap"`;
    SimplePointerArray [] * SIMPLE    `db:"-"`;
}

// ~~

type VALUE struct {
    Id int64    `db:"Id"`;
    Uuid gocql.UUID    `db:"Uuid"`;
    Name string    `db:"Name"`;
    Integer int32    `db:"Integer"`;
    Text string    `db:"Text"`;
}

// ~~

type VALUE_DATA struct {
    Uuid gocql.UUID    `db:"Uuid"`;
    ValueId int64    `db:"ValueId"`;
    ValueUuid gocql.UUID    `db:"ValueUuid"`;
    InternalId int64    `db:"-"`;
    InternalUuid gocql.UUID    `db:"-"`;
    InternalValueId int64    `db:"-"`;
    InternalValueUuid gocql.UUID    `db:"-"`;
}

// ~~

type INTERNAL_DATA struct {
    Id int64    `db:"Id"`;
    Uuid gocql.UUID    `db:"-"`;
    ValueId int64    `db:"ValueId"`;
    ValueUuid gocql.UUID    `db:"ValueUuid"`;
    FirstName string    `db:"-"`;
    LastName string    `db:"-"`;
}

File: TEST/CS/test_type.cs
public class SIMPLE
{
    public Guid
        Uuid;
    public bool
        Bool;
    public sbyte
        Int8;
    public byte
        Uint8;
    public int
        Int16;
    public uint
        Uint16;
    public int
        Int32;
    public uint
        Uint32;
    public long
        Int64;
    public ulong
        Uint64;
    public float
        Float32;
    public double
        Float64;
    public String
        String,
        Name;
    public DateTime
        Date,
        DateTime;
    public String
        Blob;
}

// ~~

public class COMPOUND
{
    public int
        Id;
    public String
        Location;
    public Tuple< String, String >
        Name;
    public HashSet< Tuple< String, String > >
        NameSet;
    public List< String >
        PhoneList;
    public HashSet< String >
        EmailSet;
    public Dictionary< String, String >
        CompanyMap;
    public DateTime
        SimpleDate;
    public Dictionary< Tuple< String, String >, DateTime >
        SimpleDateMap;
    public HashSet< DateTime >
        SimpleDateSet;
    public List< DateTime >
        SimpleDateList;
    public Dictionary< DateTime, HashSet< Tuple< String, String > > >
        NameSetMap;
    public String
        Text;
    public List< SIMPLE >
        SimplePointerArray;
}

// ~~

public class VALUE
{
    public long
        Id;
    public Guid
        Uuid;
    public String
        Name;
    public int
        Integer;
    public String
        Text;
}

// ~~

public class VALUE_DATA
{
    public Guid
        Uuid;
    public long
        ValueId;
    public Guid
        ValueUuid;
    public long
        InternalId;
    public Guid
        InternalUuid;
    public long
        InternalValueId;
    public Guid
        InternalValueUuid;
}

// ~~

public class INTERNAL_DATA
{
    public long
        Id;
    public Guid
        Uuid;
    public long
        ValueId;
    public Guid
        ValueUuid;
    public String
        FirstName,
        LastName;
    public bool
        IsAwesome;
}
